daemon/exec.go

```diff
@@ -249,6 +250,8 @@ func (d *Daemon) ContainerExecStart(ctx context.Context, name string, stdin io.R
 	ec.Lock()
 	c.ExecCommands.Lock()
 	systemPid, err := d.containerd.Exec(ctx, c.ID, ec.ID, p, cStdin != nil, ec.InitializeStdio)
+	// the exec context should be ready, or error happened.
+	ec.MakeReady()
```

Comment:
just close the channel directly, no need to check for non nil if it's initialized in `NewConfig()` as suggested in my other comment

Reply:
copy that~